baybe/surrogates/base.py

```diff
@@ -96,47 +95,22 @@ def transform_targets(self, data: pd.DataFrame) -> pd.DataFrame:
             raise ModelNotTrainedError("The model must be trained first.")
         return self._target_transform(data)
 
-    def posterior(self, candidates: pd.DataFrame) -> tuple[Tensor, Tensor]:
+    def posterior(self, candidates: pd.DataFrame) -> Posterior:
         """Evaluate the surrogate model at the given candidate points."""
-        import torch
-
-        # Evaluate the posterior distribution
-        mean, covar = self._posterior(to_tensor(self.transform_inputs(candidates)))
-
-        # Apply covariance transformation for marginal posterior models
-        if not self.joint_posterior:
-            # Convert to tensor containing covariance matrices
-            covar = torch.diag_embed(covar)
-
-        # Add small diagonal variances for numerical stability
-        covar.add_(torch.eye(covar.shape[-1]) * _MIN_VARIANCE)
-
-        return mean, covar
+        return self._posterior(to_tensor(self.transform_inputs(candidates)))
 
     @abstractmethod
-    def _posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-        """Perform the actual posterior evaluation logic.
-
-        In contrast to its public counterpart
-        :func:`baybe.surrogates.Surrogate.posterior`, no data
-        validation/transformation is carried out but only the raw posterior computation
-        is conducted.
-
-        Note that the public ``posterior`` method *always* returns a full covariance
-        matrix. By contrast, this method may return either a covariance matrix or a
-        tensor of marginal variances, depending on the models ``joint_posterior``
-        flag. The optional conversion to a covariance matrix is handled by the public
-        method.
-
-        See :func:`baybe.surrogates.Surrogate.posterior` for details on the
-        parameters.
+    def _posterior(self, candidates: Tensor) -> Posterior:
+        """Perform the actual posterior evaluation logic."""
```

Comment:
Fully agree, and this will definitely come, not only for `posterior` but also for the `fit` methods, etc. But the I did not add it yet, because the entire functions and their signatures are still WIP (i.e., the last PR changed the input arguments, this PR changed the output, and the upcoming PRs might still abstract away the torch backend / add additional steps). So I'd like to do it at the very end. Is it OK if we just keep this in mind, and you can also help me not to forget it? Everything I could write at this moment will very likely need to be replaced again in the end, so I'd prefer to save some work in the PRs and rather progress as fast as possible.

Reply:
I will never not complain about docstrings that I feel are not precise enough, don't worry 😉